scalding-beam/src/main/scala/com/twitter/scalding/beam_backend/BeamWriter.scala

```diff
@@ -48,7 +75,16 @@ class BeamWriter(val beamMode: BeamMode) extends Writer {
               }
               rec(xs)
             }
-            //TODO: handle Force and ToIterable
+            case OptimizedWrite(pipe, ToWrite.ToIterable(opt)) =>
+              val pcoll = planner(opt).run(pipeline)
+              val tempLocation = pcoll.getPipeline.getOptions.getTempLocation
+              assert(tempLocation != null, "Temp location cannot be null when using toIterableExecution")
```

Comment:
this should probably be a `require`. An assertion is something that *must* be true by construction, not a requirement.